lib/broadway/topology.ex

```diff
@@ -151,12 +151,16 @@ defmodule Broadway.Topology do
       processors_config: init_processors_config(opts[:processors]),
       batchers_config: opts[:batchers],
       context: opts[:context],
-      terminator: :"#{name_prefix(opts[:name])}.Terminator",
       max_restarts: opts[:max_restarts],
       max_seconds: opts[:max_seconds],
       shutdown: opts[:shutdown],
       resubscribe_interval: opts[:resubscribe_interval]
     }
+    |> put_terminator()
+  end
+
+  defp put_terminator(config) do
+    Map.put(config, :terminator, process_name(config, "Terminator"))
```

Comment:
We need to wait until `config` is built before calling `process_name`, which takes the config.

The alternative is to change `process_name` from this:

```
process_name(config, "Terminator")
```

To this:

```
process_name(config.module, config.name, "Terminator"
```

Where the module and broadway name are extracted from the config first then passed down to `process_name`. Happy to go that route if desired

Reply:
@josevalim You move fast! 😂 While you were reviewing, I switched from `process_name(config.module, config.name, "Terminator")` -> `process_name(config, "Terminator")`, but happy to switch back

Reply:
I like `process_name/2` too.